src/plugins/log_files/log_files_impl.cpp

```diff
@@ -203,6 +203,21 @@ void LogFilesImpl::list_timeout()
     }
 }
 
+LogFiles::Result LogFilesImpl::download_log_file(LogFiles::Entry entry, const std::string& file_path)
+{
+    auto prom =
+        std::make_shared<std::promise<LogFiles::Result>>();
+    auto future_result = prom->get_future();
+
+    download_log_file_async(entry, file_path, [prom](LogFiles::Result result, LogFiles::ProgressData progress) {
+        UNUSED(progress);
+        if (result != LogFiles::Result::Next) {
+            prom->set_value(result);
+	}
+    });
+    return future_result.get();
```

Comment:
This is fine, thanks for that. To enable it in the API you can remove this line here: https://github.com/mavlink/MAVSDK-Proto/blob/11769a4ee6f3f9b0fef71532f773d40ab3e7d55c/protos/log_files/log_files.proto#L17